binhollc/SupernovaController
# #1237 -  Integrate GPIO Functionality into SupernovaController

Resolves [1237](https://focusuy.atlassian.net/browse/BMC2-1237).

## How to test

0. Run `pip install .`
1. Shortcut pin 5 with pin 6.
2. Run `python examples\basic_gpio_example.py`.
3. Verify gpio tests don't fail: `python tests\test.py`.

## Expected results

```
Opening Supernova host adapter device and getting access to the GPIO interface...
Initializing GPIO peripheral.
Configuring GPIO pins...
Setting GPIO_6 to HIGH and reading value from GPIO_5...
GPIO_5 read value: HIGH
Setting GPIO_6 to LOW and reading value from GPIO_5...
GPIO_5 read value: LOW
Registering GPIO interrupt handlers...
Configuring interrupt on GPIO_5 for both rising and falling edges...
Toggling GPIO_6 to generate interrupts on GPIO_5...
GPIO interrupt notification on pin: GPIO_5
GPIO interrupt notification on pin: GPIO_5
GPIO interrupt notification on pin: GPIO_5
GPIO interrupt notification on pin: GPIO_5
Disabling interrupt on GPIO_5...
Closing Supernova device connection...
```

## Documentation

The design decision regarding bus voltage setting procedure is documented in [this comment](https://focusuy.atlassian.net/browse/BMC2-1237?focusedCommentId=15934).
